x-pack/test/functional/page_objects/security_page.ts

```diff
@@ -400,104 +397,78 @@ export function SecurityPageProvider({ getService, getPageObjects }: FtrProvider
-                });
-            }
-          }) // clicking save button
-          .then(async () => {
-            log.debug('click save button');
-            await testSubjects.click('roleFormSaveButton');
-          })
-          .then(function () {
-            return PageObjects.common.sleep(5000);
-          })
-      );
+      await this.clickNewRole();
+
+      // We have to use non-test-subject selectors because this markup is generated by ui-select.
+      log.debug('roleObj.indices[0].names = ' + roleObj.elasticsearch.indices[0].names);
+      await testSubjects.append('roleFormNameInput', roleName);
+
+      for (const indexName of roleObj.elasticsearch.indices[0].names) {
+        await comboBox.setCustom('indicesInput0', indexName);
+      }
+
+      if (roleObj.elasticsearch.indices[0].query) {
+        await testSubjects.click('restrictDocumentsQuery0');
+        await testSubjects.setValue('queryInput0', roleObj.elasticsearch.indices[0].query);
+      }
+
+      const globalPrivileges = (roleObj.kibana as any).global;
+      if (globalPrivileges) {
+        for (const privilegeName of globalPrivileges) {
+          await testSubjects.click('addSpacePrivilegeButton');
+
+          await testSubjects.click('spaceSelectorComboBox');
+
+          const globalSpaceOption = await find.byCssSelector(`#spaceOption_\\*`);
+          await globalSpaceOption.click();
+
+          await testSubjects.click('basePrivilegeComboBox');
+
+          const privilegeOption = await find.byCssSelector(`#basePrivilege_${privilegeName}`);
+          await privilegeOption.click();
+
+          await testSubjects.click('createSpacePrivilegeButton');
+        }
+      }
+
+      function addPrivilege(privileges: string[]) {
+        return privileges.reduce(function (promise: Promise<any>, privilegeName: string) {
+          return promise
+            .then(() => self.addPrivilegeToRole(privilegeName))
+            .then(() => PageObjects.common.sleep(250));
+        }, Promise.resolve());
+      }
+
+      await addPrivilege(roleObj.elasticsearch.indices[0].privileges);
+
+      async function addGrantedField(fields: string[]) {
+        for (const entry of fields) {
+          await comboBox.setCustom('fieldInput0', entry);
+        }
+      }
+
+      // clicking the Granted fields and removing the asterix
+      if (roleObj.elasticsearch.indices[0].field_security) {
+        // Toggle FLS switch
+        await testSubjects.click('restrictFieldsQuery0');
+
+        // have to remove the '*'
+        await find.clickByCssSelector(
+          'div[data-test-subj="fieldInput0"] [title="Remove * from selection in this group"] svg.euiIcon'
+        );
+
+        await addGrantedField(roleObj.elasticsearch.indices[0].field_security!.grant!);
+      }
+
+      log.debug('click save button');
+      await testSubjects.click('roleFormSaveButton');
+
+      // Signifies that the role managemet page redirected back to the role grid page,
```

Comment:
Supernit

```suggestion
      // Signifies that the role management page redirected back to the role grid page,
```

Reply:
> I see that you added and then reverted a couple of commits, do you have any comments on those for future reference?

Yeah I can update the PR description. The tests need to be rewritten as API tests (potentially). They aren't suited as jest tests, and end up being super flakey